packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -91,7 +92,7 @@ export async function handleHotUpdate(
 	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
 	// These will cause full reloads, so filter them out here
 	const mods = [...filtered].filter((m) => !m.url.endsWith('='));
-	const file = ctx.file.replace(config.root.pathname, '/');
+	const file = relative(process.cwd(), ctx.file.replace(config.root.pathname, '/'));
```

Comment:
I think involving cwd might create unexpected results. Most users would start the dev server from the root of their projects, but they should not see something different if that is not the case.

Reply:
could `import.meta.url` help?

Reply:
`import.meta.url` would likely be somewhere in node_modules.

It also seems like `relative()`, and `ctx.file.replace()` are attempting to do the same thing: removing the first few parts of ctx.file.

Why doesn't the `replace()` call work on Windows?

Reply:
Pushing a fix soon but here is what happens:
```ts
{
  pathname: '/D:/DEV/PROJECTS/withastro/astro/examples/basics/',
  file: 'D:/DEV/PROJECTS/withastro/astro/packages/astro/dist/jsx-runtime/index.js'
}
```
the only thing is to prepend a slash if needed